yjit/src/backend/ir.rs

```diff
@@ -467,6 +470,9 @@ pub enum Insn {
     /// Shift a value right by a certain amount (signed).
     RShift { opnd: Opnd, shift: Opnd, out: Opnd },
 
+    /// Set a side exit context for subsequent instructions.
+    SideExitContext { pc: *mut VALUE, stack_size: u8 },
```

Comment:
I wonder if this really needs to be an instruction? Could potentially call `asm.set_side_exit_context()` and set some mutable fields in the `Assembler` instead? That would avoid allocating and copying these pseudo-instructions?

Reply:
This is similar to https://github.com/ruby/ruby/pull/7712#discussion_r1166913625. If we were to introduce another iterator for `Target` and convert all `Target::SideExit` to `Target::SideExitPr` on `push_insn`, having only a single mutable field would be enough, but it's generally too late at the time of `asm.compile`.

However, assuming we shouldn't use `CodegenGlobals::get_outlined_cb()` at this place, resolving a `Target` on `push_insn` would require `ocb` for every `Assembler` API that takes `Target`s, e.g. `asm.je(target, ocb)`. And I wasn't sure if we want that API.

Reply:
Presumably, instead of inserting `Insn::SideExitContext { pc, stack_size }` pseudo-instructions, you could store those values on the `Assembler`, and when a `Target::SideExit` is inserted, you would write those values into the target?

Reply:
Sounds like a good idea. I didn't think about adding an iterator for `Target`s, but it might be useful for what's discussed in https://github.com/ruby/ruby/pull/7712#discussion_r1167092787 too. I'll have a look at removing this instruction.

Reply:
removed `Insn::SideExitContext` at https://github.com/ruby/ruby/pull/7712/commits/a61a4cb7adcd78a307eb7e4e728f611ed0aeabfd.